gdk/runtime_config.py

```diff
@@ -0,0 +1,102 @@
+import json
+import logging
+import os
+from enum import Enum
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigKey(Enum):
+    INSTALLED = 'installed'
+
+
+class Singleton(type):
+    """
+    Metaclass makes a class a singleton.
+    """
+
+    def __init__(cls, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        cls.__instance = None
+
+    def __call__(cls, *args, **kwargs):
+        """
+        Called every time a new instance of a class is being implemented.
+        """
+        force_create = kwargs.get('force_create', False)
+
+        if not cls.__instance or force_create is True:
+            cls.__instance = super().__call__(*args, **kwargs)
+
+        return cls.__instance
+
+
+GDK_RUNTIME_CONFIG_DIR = "__GDK_RUNTIME_CONFIG_PATH"
+
+
+class RuntimeConfig(metaclass=Singleton):
+    """
+    Wrapper around a dict that persists values used by the application by flushing them to
+    disk.
+    """
+    DEFAULT_CONFIG_FILE_NAME = 'runtime.json'
```

Comment:
If this is only telemetry related, we can give it some other name like `telemtry.json`. If not, `runtime.json` works.